e2e/cat_test.go

```diff
@@ -70,67 +70,59 @@ func TestCatS3Object(t *testing.T) {
 func TestCatS3ObjectFail(t *testing.T) {
 	t.Parallel()
 
-	const (
-		bucket   = "bucket"
-		filename = "file.txt"
-	)
-
-	bucketSrc := fmt.Sprintf("s3://%v", bucket)
-	prefixSrc := fmt.Sprintf("%v/prefix", bucketSrc)
-	src := fmt.Sprintf("%s/%v", prefixSrc, filename)
-
 	testcases := []struct {
+		src       string
 		name      string
 		cmd       []string
 		expected  map[int]compareFunc
 		assertOps []assertOp
 	}{
 		{
+			src:  "s3://%v/prefix/file.txt",
 			name: "cat non existent remote object",
 			cmd: []string{
 				"cat",
-				src,
 			},
 			expected: map[int]compareFunc{
-				0: contains(`ERROR "cat s3://bucket/prefix/file.txt": NoSuchKey: status code: 404`),
+				0: match(`ERROR "cat s3://(.*)/prefix/file\.txt": NoSuchKey:`),
 			},
 		},
 		{
+			src:  "s3://%v/prefix/file.txt",
 			name: "cat non existent remote object with json flag",
 			cmd: []string{
 				"--json",
 				"cat",
-				src,
 			},
 			expected: map[int]compareFunc{
-				0: contains(`{"operation":"cat","command":"cat s3://bucket/prefix/file.txt","error":"NoSuchKey: status code: 404,`),
+				0: match(`{"operation":"cat","command":"cat s3:\/\/(.*)\/prefix\/file\.txt","error":"NoSuchKey:`),
 			},
 			assertOps: []assertOp{
 				jsonCheck(true),
 			},
 		},
 		{
+			src:  "s3://%v/prefix/file.txt/*",
 			name: "cat remote object with glob",
 			cmd: []string{
 				"--json",
 				"cat",
-				src + "/*",
 			},
 			expected: map[int]compareFunc{
-				0: equals(`{"operation":"cat","command":"cat s3://bucket/prefix/file.txt/*","error":"remote source \"s3://bucket/prefix/file.txt/*\" can not contain glob characters"}`),
+				0: match(`{"operation":"cat","command":"cat s3:\/\/(.*)\/prefix\/file\.txt\/\*","error":"remote source \\"s3:\/\/(.*)\/prefix\/file\.txt\/\*\\" can not contain glob characters"}`),
```

Comment:
Let's make pattern that matches the bucket name non greedy, and make sure it matches at least one character
```suggestion
				0: match(`{"operation":"cat","command":"cat s3:\/\/(.+)?\/prefix\/file\.txt\/\*","error":"remote source \\"s3:\/\/(.*)\/prefix\/file\.txt\/\*\\" can not contain glob characters"}`),
```